index.js

```diff
@@ -345,24 +347,41 @@ Client.prototype._encode = function (obj, enc) {
   return ndjson.serialize(out)
 }
 
+Client.prototype._flushCallsWaitingMetadata = function () {
+  for (const args of this._callsWaitingMetadata) {
+    this.write(...args)
+  }
+
+  this._callsWaitingMetadata = []
+}
+
+Client.prototype._sendOrAddToWaitingMetadata = function(...args) {
+  if (this._hasSentMetadata) {
+    this._maybeUncork()
+    return this.write(...args)
+  } else {
+    if (this._callsWaitingMetadata.length === 0) {
+      this._chopper.write('')
```

Comment:
This was basically to trigger the metadata to be written first, as it would only ever go through this code path if this is the request that would cause a new stream to be created (`this._hasSentMetadata` is `true` and `this._callsWaitingMetadata` is empty).

We cannot use cork for that as it's not possible to force write some data to the stream while it's corked.

If there is a better way to do this, please let me know.


Return value on sendSpan / sendTransaction is always true when waiting for the metadata to be sent, this has no implications on the current code base as those return values are ignored.


Called write('') to trigger streamchopper